br/pkg/lightning/backend/external/codec.go

```diff
@@ -0,0 +1,78 @@
+// Copyright 2023 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package external
+
+import "encoding/binary"
+
+type rangeProperty struct {
+	key    []byte
+	offset uint64
+	size   uint64
+	keys   uint64
+}
+
+// decodeMultiProps is only used for test.
+func decodeMultiProps(data []byte) ([]*rangeProperty, error) {
+	var ret []*rangeProperty
+	for len(data) > 0 {
+		propLen := int(binary.BigEndian.Uint32(data))
+		propBytes := data[4 : 4+propLen]
```

Comment:
protect out-of-boundary

Reply:
I prefer to let it panic to keep the code clean.

Reply:
Will it affect other SQL connection? Or we have a recover at caller to handle it